providers/provider_data.go

```diff
@@ -13,6 +13,7 @@ type ProviderData struct {
 	ProfileURL        *url.URL
 	ProtectedResource *url.URL
 	ValidateURL       *url.URL
+	GroupsURL         *url.URL
```

Comment:
This type of Group API is unique to Azure, isn't it? I don't think it should live in ProviderData if that's the case.